src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py

```diff
@@ -100,33 +101,62 @@ def load_config(self):
         else:
             self.metadata_mgr = MetadataManager(self.fs, self.config_path, 0o640)
 
-    def set_attrs(self, path, size, isolate_namespace, pool, uid, gid):
+    def get_attrs(self, pathname):
+        # get subvolume attributes
+        attrs = {}
+        stx = self.fs.statx(pathname,
+                            cephfs.CEPH_STATX_UID | cephfs.CEPH_STATX_GID | cephfs.CEPH_STATX_MODE,
+                            cephfs.AT_SYMLINK_NOFOLLOW)
+
+        attrs["uid"] = int(stx["uid"])
+        attrs["gid"] = int(stx["gid"])
+        attrs["mode"] = int(int(stx["mode"]) & ~stat.S_IFMT(stx["mode"]))
+
+        try:
+            attrs["data_pool"] = self.fs.getxattr(pathname, 'ceph.dir.layout.pool').decode('utf-8')
+        except cephfs.NoData:
+            attrs["data_pool"] = None
+
+        try:
+            attrs["pool_namespace"] = self.fs.getxattr(pathname, 'ceph.dir.layout.pool_namespace').decode('utf-8')
+        except cephfs.NoData:
+            attrs["pool_namespace"] = None
+
+        try:
+            attrs["quota"] = int(self.fs.getxattr(pathname, 'ceph.quota.max_bytes').decode('utf-8'))
+        except cephfs.NoData:
+            attrs["quota"] = None
+
+        return attrs
+
+    def set_attrs(self, path, attrs):
```

Comment:
> Do you plan to call this in cloner?

Not in this PR or as yet. Just creating a cleaner interface as you suggested, for now.

Reply:
+1 if it can be included in this PR.

Reply:
> general comment: pythonic way of writing this would be:
> 
> ```
> if attrs.get(key, None):
>    ....
> ```
> 
> this way, the routine can handle a subset of all supported attrs.

Addressed the above.

Reply:
> +1 if it can be included in this PR.

@vshankar I intend to pick this up when addressing https://tracker.ceph.com/issues/46399 (preserve xattrs during snapshot clone), which is about 2 issues down the line in my queue.

As at that point in time we would add get/set xattrs as well to these functions and use the common methods in both setting up the subvolume root and any directories and files within it.

Let me know if this works for you.

Reply:
ack -- works for me.